x/concentrated-liquidity/lp.go

```diff
@@ -244,3 +249,48 @@ func (k Keeper) initializeInitialPosition(ctx sdk.Context, pool types.Concentrat
 	}
 	return nil
 }
+
+// GetSecondsPerLiquidityInside returns the seconds per liquidity between two ticks in a given pool
+func (k Keeper) GetSecondsPerLiquidityInside(ctx sdk.Context, poolId uint64, lowerTick, upperTick int64) (sdk.Dec, error) {
+	// Get the Seconds Per Liquidity Outside for the lower tick
+	lowerTickInfo, err := k.getTickInfo(ctx, poolId, lowerTick)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+
+	// Get the Seconds Per Liquidity Outside for the upper tick
+	upperTickInfo, err := k.getTickInfo(ctx, poolId, upperTick)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+
+	pool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+
+	globalSecondsPerLiquidity := pool.GetGlobalSecondsPerLiquidity()
+
+	// If tick is active, we need to calculate a new seconds per liquidity inside since we only update these when crossing ticks
+	if pool.GetCurrentTick().GTE(sdk.NewInt(lowerTick)) || pool.GetCurrentTick().LT(sdk.NewInt(upperTick)) {
```

Comment:
Did you mean if the _range_ is active here? Also if so, what's the reasoning behind having the upper tick check be non-inclusive (`LT`)?

Reply:
Yes, I meant range is active. As for the upper tick being non inclusive, this is the way initialized tick searching was designed and would have to dig deeper to further understand. See:

https://github.com/osmosis-labs/osmosis/blob/45d143fa83100d8f985ea2cd65f9210d027d3b2a/x/concentrated-liquidity/internal/swapstrategy/zero_for_one.go#L77-L87

and

https://github.com/osmosis-labs/osmosis/blob/45d143fa83100d8f985ea2cd65f9210d027d3b2a/x/concentrated-liquidity/internal/swapstrategy/one_for_zero.go#L78-L89